### ISSUE TITLE
More verbosity when re-initializing components

### ISSUE THREAD

**Issue by ottonemo** [dnouri/skorch#349]
I often continue training my models using `partial_fit` preceded by a `set_params()` where I increase/decrease parameters such as the learning rate. However, certain parameter modifications trigger a re-initialization, e.g. setting the learning rate using `set_params` will re-initialize the optimizer. This can be problematic for optimizers that have a state, such as `RMSprop` or `Adam`.

So either 
- we enable the user to set parameters such as the learning rate without re-initializing the optimizer or
- we make this initialization visible.

**Comment by BenjaminBossan** [comment]
> we enable the user to set parameters such as the learning rate without re-initializing the optimizer

Is there a reliable way to do this? My impression is that some parameters such as learning rate are treated differently by the optimizer, so that a simple `setattr` isn't actually enough. Maybe we should advise users that the safest way to achieve these kinds of things is to use learning rate schedulers, since they *know* what needs to be done.

**Comment by vykhand** [comment]
@BenjaminBossan I believe we need a reliable way to reload an optimizer. Changing LR is not the only use case for this. I am trying to restart model training with a different loss function, and currently it does not work.

**Comment by ottonemo** [comment]
> @BenjaminBossan I believe we need a reliable way to reload an optimizer. Changing LR is not the only use case for this. I am trying to restart model training with a different loss function, and currently it does not work.

But that should be already possible. You will lose the optimizer's state (e.g., for Adam the running average estimation of the FI matrix) but in your case this is probably OK. Example:
```python
net.set_params(optimizer=torch.optim.Adam, criterion=torch.nn.NLLLoss)
```
should work fine.
